src/components/dock/dock.tsx

```diff
@@ -0,0 +1,187 @@
+import {
+    Component,
+    Element,
+    Event,
+    EventEmitter,
+    h,
+    Host,
+    Prop,
+    State,
+} from '@stencil/core';
+import { DockItem } from './dock.types';
+
+const DEFAULT_MOBILE_BREAKPOINT = 700;
+
+/**
+ * @exampleComponent limel-example-dock-basic
+ * @exampleComponent limel-example-dock-custom-component
+ * @exampleComponent limel-example-dock-mobile
+ * @exampleComponent limel-example-dock-expanded
+ * @exampleComponent limel-example-dock-colors
+ * @private
+ */
+@Component({
+    tag: 'limel-dock',
+    shadow: true,
+    styleUrl: 'dock.scss',
+})
+export class Dock {
+    @Element()
+    public host: HTMLLimelDockElement;
+
+    /**
+     * Items that are placed within the `<nav />` section of the dock.
+     */
+    @Prop()
+    public dockItems: DockItem[] = [];
+
+    /**
+     * Fired when a Dock item has been selected from the dock.
+     */
+    @Event()
+    public change: EventEmitter<DockItem>;
```

Comment:
Stencil warns about using any event name that conflicts with some native DOM event name. Don't we want to another name?

Reply:
I'm also thinking that we might want to emit an event with detail `DockItem['value']` instead?

Reply:
My thoughts on this has originally been that we are (or were, at the time), mostly making replacements for standard components. Like, instead of just using `<button>`, you'd use `<limel-button>` and get our styling and perhaps some nice extra functionality. As such, it would make sense for the consumer to expect the same kind of events from `<limel-x>` as from the native `<x>`. I still think that to some extent, but with the warnings not being possible to disable, and with me seeing the following warning for the first time today, a warning that has clearly been hiding in plain sight and missed because of all the warnings about overriding native event names, I'm really starting to change my mind quite rapidly… 😣

<img width="716" alt="Build Warn: ./src/components/input-field/input-field.tsx:104:12 The @Prop() name &quot;prefix&quot; is a reserved public name. Please rename the &quot;prefix&quot; prop so it does not conflict with an existing standardized prototype member. Reusing prop names that are already defined on the element's prototype may cause unexpected runtime errors or user-interface issues on various browsers, so it's best to avoid them entirely." src="https://user-images.githubusercontent.com/154725/179941968-0f27c0c9-6af5-4524-a012-c4f67f154e92.png">

Reply:
Should we prefix these with `limel` too? So `limelChange` in this case?

Since changing existing ones will be a breaking change, perhaps we should at least start prefixing any new ones we add?

Reply:
Ah, I honestly didn't look at what the event in question was in this specific instance, so I totally missed that the current name wasn't very fitting. I agree that this event should not be called `change`, regardless of any other event-naming concerns we have more generally 😄

`itemSelected` sounds good to me 👍

Reply:
I guess we need to check if using capital letters in an event name might cause issues with jsx though. I'm thinking if `onItemSelected={…}` might be interpreted to mean the event name is `item-selected`?

Reply:
Camel cased events are normal in JSX, and that's also what Stencil uses in their examples: https://stenciljs.com/docs/events

I'm not sure now if they're transformed to lowercase though. Native events are always lowercase, even if React JSX uses camel casing for them...

Reply:
Update: Camel cased custom events are not transformed to lowercase, so `onItemSelected` is same as listening to `itemSelected`.

Native event names also need to be camel cased in JSX, but are transformed to lowercase (with Stencil, you will get a type error if you try to use `onTransitionend` instead of `onTransitionEnd`.

Reply:
> If we avoid single words for our own events, we will be fine.

Appears I was wrong here, at least when using JSX in Stencil. If I define a custom event with

```ts
@Event()
    public transitionEnd: EventEmitter;
```

I will get a build warning about event name conflict (if I also use that event emitter). But the emitted event will still be `transitionEnd` which is not really conflicting with the native DOM event when using `element.addEventListener`, etc. But there is a conflict when using JSX, so you will end up listening to the native event instead!

Still, we're less likely to run into event name conflicts when using camel cased event names because most native DOM events are single words. But then we really _have to_ avoid those name conflicts, otherwise it may be impossible to bind an event listener using JSX.

Reply:
⚡

Event name changed from `selected` to `itemSelected`.